build: cypress build no longer optional?

**Describe the problem**

I saw the 22.2 release announcement (congrats!) and went to build an OSS build on illumos.  I ran into this failure:

```
error /home/dap/garbage-compactor/cockroach/cache/gopath/src/github.com/cockroachdb/cockroach/pkg/ui/workspaces/e2e-tests/node_modules/cypress: Command failed.
Exit code: 1
Command: node index.js --exec install
Arguments: 
Directory: /home/dap/garbage-compactor/cockroach/cache/gopath/src/github.com/cockroachdb/cockroach/pkg/ui/workspaces/e2e-tests/node_modules/cypress
Output:
Platform: "sunos" is not supported.
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
gmake: *** [Makefile:371: pkg/ui/yarn.installed] Error 1
gmake: *** Waiting for unfinished jobs....
```

Is this a regression of #47694?

I applied this patch:

```
index c93a67e..be5518c 100755
--- a/src/github.com/cockroachdb/cockroach/pkg/ui/not-yarn-workspace.sh
+++ b/src/github.com/cockroachdb/cockroach/pkg/ui/not-yarn-workspace.sh
@@ -7,7 +7,7 @@ set -euo pipefail
   yarn --cwd workspaces/db-console/src/js install
   yarn --cwd workspaces/cluster-ui install
   yarn --cwd workspaces/db-console install
-  yarn --cwd workspaces/e2e-tests install
+  # yarn --cwd workspaces/e2e-tests install
 )
 
 cat << "EOF"
```

and was able to complete a build.

Let me know if there's more information you need.  I removed the other template sections from this report because they seemed not relevant.)

Jira issue: CRDB-22195

--- a/WORKSPACE
+++ b/WORKSPACE
@@ -280,6 +280,17 @@ yarn_install(
     strict_visibility = False,
     yarn_lock = "//pkg/ui/workspaces/e2e-tests:yarn.lock",
     symlink_node_modules = True,
+    environment = {
+        # Don't automatically install the native Cypress binary, since not all
+        # platforms that build CRDB have Cypress binaries to install:
+        # https://docs.cypress.io/guides/getting-started/installing-cypress#System-requirements
+        #
+        # The native binary will be installed by `./dev ui e2e` just-in-time.
+        # While unsupported platforms will still encounter errors at that
+        # point, UI end-to-end tests aren't part of the core build or test
+        # flows and are intended for regression testing by CRDB developers.
+        "CYPRESS_INSTALL_BINARY": "0",
+    }
 )
 
 yarn_install(

--- a/pkg/cmd/dev/testdata/datadriven/ui
+++ b/pkg/cmd/dev/testdata/datadriven/ui
@@ -153,6 +153,7 @@ bazel info workspace --color=no
 bazel info workspace --color=no
 bazel run @yarn//:yarn -- --silent --cwd crdb-checkout/pkg/ui/workspaces/e2e-tests install
 bazel build //pkg/cmd/cockroach:cockroach --config=with_ui
+bazel run @yarn//:yarn -- --silent --cwd crdb-checkout/pkg/ui/workspaces/e2e-tests cypress install
 crdb-checkout/pkg/ui/workspaces/e2e-tests/build/start-crdb-then.sh bazel run @yarn//:yarn -- --silent --cwd crdb-checkout/pkg/ui/workspaces/e2e-tests cy:run
 
 exec
@@ -162,6 +163,7 @@ bazel info workspace --color=no
 bazel info workspace --color=no
 bazel run @yarn//:yarn -- --silent --cwd crdb-checkout/pkg/ui/workspaces/e2e-tests install
 bazel build //pkg/cmd/cockroach:cockroach --config=with_ui
+bazel run @yarn//:yarn -- --silent --cwd crdb-checkout/pkg/ui/workspaces/e2e-tests cypress install
 crdb-checkout/pkg/ui/workspaces/e2e-tests/build/start-crdb-then.sh bazel run @yarn//:yarn -- --silent --cwd crdb-checkout/pkg/ui/workspaces/e2e-tests cy:run ./foo/bar
 
 exec
@@ -171,4 +173,5 @@ bazel info workspace --color=no
 bazel info workspace --color=no
 bazel run @yarn//:yarn -- --silent --cwd crdb-checkout/pkg/ui/workspaces/e2e-tests install
 bazel build //pkg/cmd/cockroach:cockroach --config=with_ui
+bazel run @yarn//:yarn -- --silent --cwd crdb-checkout/pkg/ui/workspaces/e2e-tests cypress install
 crdb-checkout/pkg/ui/workspaces/e2e-tests/build/start-crdb-then.sh bazel run @yarn//:yarn -- --silent --cwd crdb-checkout/pkg/ui/workspaces/e2e-tests cy:debug

--- a/pkg/cmd/dev/ui.go
+++ b/pkg/cmd/dev/ui.go
@@ -576,6 +576,14 @@ launching test in a real browser. Extra flags are passed directly to the
 				return fmt.Errorf("unable to build cockroach with UI: %w", err)
 			}
 
+			// Ensure the native Cypress binary is installed.
+			cyInstallArgv := buildBazelYarnArgv("--silent", "--cwd", uiDirs.e2eTests, "cypress", "install")
+			logCommand("bazel", cyInstallArgv...)
+			err = d.exec.CommandContextInheritingStdStreams(ctx, "bazel", cyInstallArgv...)
+			if err != nil {
+				return fmt.Errorf("unable to install Cypress native package: %w", err)
+			}
+
 			// Run Cypress tests, passing any extra args through to 'cypress'
 			startCrdbThenSh := path.Join(uiDirs.e2eTests, "build/start-crdb-then.sh")
 			runCypressArgv := append(

--- a/pkg/ui/not-yarn-workspace.sh
+++ b/pkg/ui/not-yarn-workspace.sh
@@ -7,7 +7,8 @@ set -euo pipefail
   yarn --cwd workspaces/db-console/src/js install
   yarn --cwd workspaces/cluster-ui install
   yarn --cwd workspaces/db-console install
-  yarn --cwd workspaces/e2e-tests install
+  # Don't install the native Cypress binary immediately. ./dev ui e2e will do that just in time.
+  CYPRESS_INSTALL_BINARY=0 yarn --cwd workspaces/e2e-tests install
 )
 
 cat << "EOF"

